WebApiLab.DAL/Entities/Product.cs

```diff
@@ -1,29 +1,23 @@
-﻿using System;
-using System.Collections.Generic;
-using System.ComponentModel.DataAnnotations.Schema;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-
-namespace WebApiLab.DAL.Entities
+﻿using System.ComponentModel.DataAnnotations.Schema;
+
+namespace WebApiLab.Dal.Entities;
+
+public class Product
 {
-    public class Product
-    {
-        public int Id { get; set; }
+    public int Id { get; set; }
 
-        [Column("ProductName")]
-        public string Name { get; set; }
-        public int UnitPrice { get; set; }
+    [Column("ProductName")]
+    public string Name { get; set; } = null!;
 
-        public int CategoryId { get; set; }
-        public Category Category { get; set; }
+    public int UnitPrice { get; set; }
 
-        public ICollection<OrderItem> ProductOrders { get; }
-                                        = new List<OrderItem>();
+    public int CategoryId { get; set; }
+    public Category? Category { get; set; }
```

Comment:
Ha kötelező a kapcsolat, akkor ne legyen a prop nullable, hanem null fogiving.
"At the same time, it is undesirable to make these properties nullable, since that would force all access to them to check for null, even if they are required."
https://docs.microsoft.com/en-us/ef/core/miscellaneous/nullable-reference-types

Reply:
javítva